src/Orleans/Runtime/CallbackData.cs

```diff
@@ -150,22 +153,18 @@ public void DoCallback(Message response)
 
         public void Dispose()
         {
-            DisposeTimer();
-            GC.SuppressFinalize(this);
-        }
-
-        private void DisposeTimer()
-        {
-            try
+            var timeouter = callbackTimeouter;
+            if (timeouter != null)
             {
-                var tmp = timer;
-                if (tmp != null)
+                try
                 {
-                    timer = null;
-                    tmp.Dispose();
+                    callbackTimeouter = null;
+                    timeouter.Dispose();
```

Comment:
wouldn't this dispose the timeouter for multiple CallbackData instead of just this one since this is a thread static?